sonic_y_cable/y_cable.py

```diff
@@ -507,3 +536,384 @@ def check_if_link_is_active_for_torB(physical_port):
+            if done[0] == 1:
+                break
+
+        idx = 0
+        maxLane = 2
+        curr_offset = Y_CABLE_LANE_1_BER_RESULT
+        for lane in range(maxLane):
+            msb_result = platform_chassis.get_sfp(physical_port).read_eeprom(curr_offset+idx, 1)
+            y_cable_validate_read_data(msb_result, 1, physical_port, "BER data msb result")
+            lsb_result = platform_chassis.get_sfp(physical_port).read_eeprom(curr_offset+1+idx, 1)
+            y_cable_validate_read_data(lsb_result, 1, physical_port, "BER data lsb result")
+            lane_result = msb_result[0] * math.pow(10, (lsb_result[0]-24))
+            ber_result.append(lane_result)
+            #print(' lane[%1d] BER[%e]' % ( lane + 1, lane_result))
+            idx += 2
+
+    else:
+        helper_logger.log_error("platform_chassis is not loaded, failed to configure the PRBS type")
+        return -1
+
+    return ber_result
+
+
+@hook_y_cable_simulator
+def get_eye_info(physical_port, target):
+    """
+    This API specifically returns the EYE height value for a specfic port.
+    The target could be local side, TOR1, TOR2, NIC etc.
+
+
+    Register Specification of upper page 0x5 at offset 144 is documented below
+
+    Byte offset   bits    Name                     Description
+    144           1-0     Initiate EYE Measurement "Write 0x00 - initiate gated eye height measurement on target side with any traffic (PRBS or mission mode)
+                                                    Write 0x01-0xFF - reserved
+                                                    Read 0x00 - Eye Height Measurement in process
+                                                    Read 0x01 - Eye Height Gate complete, valid values in Eye Height Registers
+                                                    NB this command can be issued whenever the side is linked, this does not interrupt mission mode traffic"	R/W
+
+    Args:
+        physical_port:
+             an Integer, the actual physical port connected to a Y cable
+        target:
+             an Integer, the target on which to enable the PRBS
+                         0 -> local side,
+                         1 -> TOR 1
+                         2 -> TOR 2
+                         3 -> NIC
+    Returns:
+        a list, with EYE values of lane 0 and lane 1 with corresponding index
+    """
+
+    buffer = bytearray([target])
+    curr_offset = Y_CABLE_TARGET
+
+    eye_result = []
+
+    if platform_chassis is not None:
+        result = platform_chassis.get_sfp(
+            physical_port).write_eeprom(curr_offset, 1, buffer)
+        if result is False:
+            return result
+        buffer = bytearray([0])
+        curr_offset = Y_CABLE_INITIATE_EYE_MEASUREMENT
+        result = platform_chassis.get_sfp(
+            physical_port).write_eeprom(curr_offset, 1, buffer)
+        if result is False:
+            return result
+
+        while(1):
```

Comment:
Suggest `while True:` to be more Pythonic.

Reply:
fixed